Sources/ProjectDescription/SettingsTransformers.swift

```diff
@@ -0,0 +1,147 @@
+        merging(with: ["CODE_SIGN_IDENTITY": identity.asSettingValue])
+    }
+
+    // MARK: - Versioning and Product Name
+
+    ///Sets "PRODUCT_NAME" to `name`
+    func productName(_ name: String) -> SettingsDictionary {
+        merging(with: ["PRODUCT_NAME": name.asSettingValue])
+    }
+
+    /// Sets "CURRENT_PROJECT_VERSION" to `version`
+    func currentProjectVersion(_ version: String) -> SettingsDictionary {
+        merging(with: ["CURRENT_PROJECT_VERSION": version.asSettingValue])
+    }
+
+    /// Sets "VERSIONING_SYSTEM" to "apple-generic"
+    func appleGenericVersioningSystem() -> SettingsDictionary {
+        merging(with: ["VERSIONING_SYSTEM": "apple-generic"])
+    }
+
+    /// Sets "VERSION_INFO_PREFIX" to `version`. If prefix is not `nil`, it's used as "VERSION_INFO_PREFIX"; and suffix as "VERSION_INFO_SUFFIX"
+    func versionInfo(_ version: String, prefix: String? = nil, suffix: String? = nil) -> SettingsDictionary {
+        var versionSettings: SettingsDictionary = ["VERSION_INFO_STRING": version.asSettingValue]
+        versionSettings["VERSION_INFO_PREFIX"] = prefix?.asSettingValue
+        versionSettings["VERSION_INFO_SUFFIX"] = suffix?.asSettingValue
+
+        return merging(with: versionSettings)
+    }
+
+    // MARK: - Swift Settings
+
+    ///Sets "SWIFT_VERSION" to `version`
+    func swiftVersion(_ version: String) -> SettingsDictionary {
+        merging(with: ["SWIFT_VERSION": version.asSettingValue])
+    }
+
+    ///Sets "OTHER_SWIFT_FLAGS" to `flags`
+    func otherSwiftFlags(_ flags: String) -> SettingsDictionary {
+        merging(with: ["OTHER_SWIFT_FLAGS": flags.asSettingValue])
+    }
+
+    /// Sets "SWIFT_COMPILATION_MODE" to the available `SwiftCompilationMode` ("singlefile" or "wholemodule")
+    func swiftCompilationMode(_ mode: SwiftCompilationMode) -> SettingsDictionary {
+        merging(with: ["SWIFT_COMPILATION_MODE": mode.rawValue.asSettingValue])
+    }
+
+    /// Sets "SWIFT_OPTIMIZATION_LEVEL" to the available `SwiftOptimizationLevel` ("-O", "-Onone" or "-Osize")
+    func swiftOptimizationLevel(_ level: SwiftOptimizationLevel) -> SettingsDictionary {
+        merging(with: ["SWIFT_OPTIMIZATION_LEVEL": level.rawValue.asSettingValue])
+    }
+
+    // MARK: - Bitcode
+
+    ///Sets "ENABLE_BITCODE" to "YES" or "NO"
+    func bitcodeEnabled(_ enabled: Bool) -> SettingsDictionary {
+        merging(with: ["ENABLE_BITCODE": enabled.asSettingValue])
+    }
+
+    // MARK: - Catalyst
+
+    ///Sets "SUPPORTS_MACCATALYST" to "YES" or "NO"
+    func supportsMacCatalist(_ enabled: Bool) -> SettingsDictionary {
+        merging(with: ["SUPPORTS_MACCATALYST": enabled.asSettingValue])
+    }
+
+    ///Sets "DERIVE_MACCATALYST_PRODUCT_BUNDLE_IDENTIFIER" to "YES" or "NO"
+    func deriveMacCatalystProductBundleId(_ enabled: Bool) -> SettingsDictionary {
+        merging(with: ["DERIVE_MACCATALYST_PRODUCT_BUNDLE_IDENTIFIER": enabled.asSettingValue])
+    }
```

Comment:
minor: typo `supportsMacCatalyst`

This another one offered via `Target.deploymentTarget` 

e.g.

```swift
Target(
   // ...
   deploymentTarget: .iOS("13.1", [.iphone, .ipad, .mac]),
   // ...
)
```

https://github.com/tuist/tuist/blob/master/Sources/TuistGenerator/Generator/ConfigGenerator.swift#L242